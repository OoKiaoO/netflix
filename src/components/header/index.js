import React from 'react';
import { Link as ReactRouterLink } from 'react-router-dom';
// Styled Components
import { Background, Container, Logo, ButtonLink } from './styles/header';

function Header({ bg = true, children, ...restProps }) {
  return bg ? <Background {...restProps}>{children}</Background> : children;
}

Header.Frame = function HeaderFrame({ children, ...restProps }) {
  return <Container {...restProps}>{children}</Container>;
};

Header.ButtonLink = function HeaderButtonLink({ children, ...restProps }) {
  return <ButtonLink {...restProps}>{children}</ButtonLink>;
};

Header.Logo = function HeaderLogo({ to, ...restProps }) {
  return (
    <ReactRouterLink to={to}>
      <Logo {...restProps} />
    </ReactRouterLink>
  );
};

export default Header;
