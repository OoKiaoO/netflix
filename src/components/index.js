// eslint-disable-next-line import/no-named-default
export { default as Jumbotron } from './jumbotron';
