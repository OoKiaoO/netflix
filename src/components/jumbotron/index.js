import React from 'react';
import { Inner, Container, Title, Subtitle, Pane, Image, Item } from './styles/jumbotron';

function Jumbotron({ children, direction = 'row', ...restProps }) {  // if no direction gets passed it will default to row
  return (
    <Item direction={direction}>
      <Inner>
        {children}
      </Inner>
    </Item>
  )
}

// compound component examples
Jumbotron.Container = function JumbotronContainer({ children, ...restProps }) {
  return <Container {...restProps}>{children}</Container>;
}

Jumbotron.Title = function JumbotronTitle({ children, ...restProps }) {
  return <Title {...restProps}>{children}</Title>;
}

Jumbotron.Subtitle = function JumbotronSubtitle({ children, ...restProps }) {
  return <Subtitle {...restProps}>{children}</Subtitle>;
}

Jumbotron.Pane = function JumbotronPane({ children, ...restProps }) {
  return <Pane {...restProps}>{children}</Pane>;
}

Jumbotron.Image = function JumbotronImage({ ...restProps }) {
  return <Image {...restProps}></Image>;
}

export default Jumbotron;
