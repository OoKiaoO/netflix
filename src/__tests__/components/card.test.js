import React from 'react';
import { render, fireEvent } from '@testing-library/react';
import { Card, Player } from '../../components';

const category = 'series';
const slideRows = [
  {
    title: 'Documentaries',
    data: [
      {
        id: '1a58ab7b-43fb-440a-af10-c5d50050968d',
        title: 'Tiger King',
        description:
          'An exploration of big cat breeding and its bizarre underworld, populated by eccentric characters.',
        genre: 'documentaries',
        maturity: '18',
        slug: 'tiger-king',
        docId: 'EoY8MRyc0kPggbKT0xzc',
      },
    ],
  },
  {
    title: 'Comedies',
    data: [
      {
        id: '21c1c541-f34c-4572-a83e-28215afd11f1',
        title: 'The Office',
        description:
          'A motley group of office workers go through hilarious misadventures at the Scranton, Pennsylvania, branch of the Dunder Mifflin Paper Company.',
        genre: 'comedies',
        maturity: '15',
        slug: 'the-office',
        docId: 'Cw5zc8sfQBQO3WOVOo2d',
      },
    ],
  },
];

describe('<Card />', () => {
  it('renders the <Card /> with populated data', () => {
    const { container, getByText } = render(
      <Card.Group>
        {slideRows.map((slideItem) => (
          <Card key={`${category}-${slideItem.title.toLowerCase()}`}>
            <Card.Title>{slideItem.title}</Card.Title>
            <Card.Entities>
              {slideItem.data.map((item) => (
                <Card.Item key={item.docId} item={item}>
                  <Card.Image src={`/images/${category}/${item.genre}/${item.slug}/small.jpg`} />
                  <Card.Meta>
                    <Card.SubTitle>{item.title}</Card.SubTitle>
                    <Card.Text>{item.description}</Card.Text>
                  </Card.Meta>
                </Card.Item>
              ))}
            </Card.Entities>
            <Card.Feature category={category}>
              <Player>
                <Player.Button />
                <Player.Video src="/videos/bunny.mp4" />
              </Player>
            </Card.Feature>
          </Card>
        ))}
      </Card.Group>
    );

    expect(getByText('Documentaries')).toBeTruthy();
    expect(getByText('Tiger King')).toBeTruthy();
    expect(
      getByText('An exploration of big cat breeding and its bizarre underworld, populated by eccentric characters.')
    ).toBeTruthy();

    expect(getByText('Comedies')).toBeTruthy();
    expect(getByText('The Office')).toBeTruthy();
    expect(
      getByText(
        'A motley group of office workers go through hilarious misadventures at the Scranton, Pennsylvania, branch of the Dunder Mifflin Paper Company.'
      )
    ).toBeTruthy();

    expect(container.firstChild).toMatchSnapshot();
  });

  it('renders the <Card /> and toggles the feature', () => {
    const { container, queryByText, getByTestId, getByAltText } = render(
      <Card.Group>
        {slideRows.map((slideItem) => (
          <Card key={`${category}-${slideItem.title.toLowerCase()}`}>
            <Card.Title>{slideItem.title}</Card.Title>
            <Card.Entities>
              {slideItem.data.map((item) => (
                <Card.Item key={item.docId} item={item} data-testid={`${item.slug}-item-feature`}>
                  <Card.Image src={`/images/${category}/${item.genre}/${item.slug}/small.jpg`} />
                  <Card.Meta>
                    <Card.SubTitle>{item.title}</Card.SubTitle>
                    <Card.Text>{item.description}</Card.Text>
                  </Card.Meta>
                </Card.Item>
              ))}
            </Card.Entities>
            <Card.Feature category={category}>
              <Player>
                <Player.Button />
                <Player.Video src="/videos/bunny.mp4" />
              </Player>
            </Card.Feature>
          </Card>
        ))}
      </Card.Group>
    );

    expect(queryByText('18')).toBeFalsy();
    fireEvent.click(getByTestId('tiger-king-item-feature'));
    expect(queryByText('18')).toBeTruthy();

    fireEvent.click(getByAltText('Close'));
    expect(queryByText('18')).toBeFalsy();

    expect(queryByText('15')).toBeFalsy();
    fireEvent.click(getByTestId('the-office-item-feature'));
    expect(queryByText('15')).toBeTruthy();

    fireEvent.click(getByAltText('Close'));
    expect(queryByText('15')).toBeFalsy();

    expect(container.firstChild).toMatchSnapshot();
  });
});
