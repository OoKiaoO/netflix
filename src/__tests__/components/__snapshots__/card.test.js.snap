// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Card /> renders the <Card /> and toggles the feature 1`] = `
<div
  class="card__Group-sc-s87lnr-2 jBSnhV"
>
  <div
    class="card__Container-sc-s87lnr-1 cVRMWf"
  >
    <p
      class="card__Title-sc-s87lnr-0 eHOFuf"
    >
      Documentaries
    </p>
    <div
      class="card__Entities-sc-s87lnr-12 kdrVnm"
    >
      <div
        class="card__Item-sc-s87lnr-13 cmuEeG"
        data-testid="tiger-king-item-feature"
      >
        <img
          class="card__Image-sc-s87lnr-14 hgMODG"
          src="/images/series/documentaries/tiger-king/small.jpg"
        />
        <div
          class="card__Meta-sc-s87lnr-11 beFYBe"
        >
          <p
            class="card__SubTitle-sc-s87lnr-3 zPluP"
          >
            Tiger King
          </p>
          <p
            class="card__Text-sc-s87lnr-4 fABboK"
          >
            An exploration of big cat breeding and its bizarre underworld, populated by eccentric characters.
          </p>
        </div>
      </div>
    </div>
  </div>
  <div
    class="card__Container-sc-s87lnr-1 cVRMWf"
  >
    <p
      class="card__Title-sc-s87lnr-0 eHOFuf"
    >
      Comedies
    </p>
    <div
      class="card__Entities-sc-s87lnr-12 kdrVnm"
    >
      <div
        class="card__Item-sc-s87lnr-13 cmuEeG"
        data-testid="the-office-item-feature"
      >
        <img
          class="card__Image-sc-s87lnr-14 hgMODG"
          src="/images/series/comedies/the-office/small.jpg"
        />
        <div
          class="card__Meta-sc-s87lnr-11 beFYBe"
        >
          <p
            class="card__SubTitle-sc-s87lnr-3 zPluP"
          >
            The Office
          </p>
          <p
            class="card__Text-sc-s87lnr-4 fABboK"
          >
            A motley group of office workers go through hilarious misadventures at the Scranton, Pennsylvania, branch of the Dunder Mifflin Paper Company.
          </p>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`<Card /> renders the <Card /> with populated data 1`] = `
<div
  class="card__Group-sc-s87lnr-2 jBSnhV"
>
  <div
    class="card__Container-sc-s87lnr-1 cVRMWf"
  >
    <p
      class="card__Title-sc-s87lnr-0 eHOFuf"
    >
      Documentaries
    </p>
    <div
      class="card__Entities-sc-s87lnr-12 kdrVnm"
    >
      <div
        class="card__Item-sc-s87lnr-13 cmuEeG"
      >
        <img
          class="card__Image-sc-s87lnr-14 hgMODG"
          src="/images/series/documentaries/tiger-king/small.jpg"
        />
        <div
          class="card__Meta-sc-s87lnr-11 beFYBe"
        >
          <p
            class="card__SubTitle-sc-s87lnr-3 zPluP"
          >
            Tiger King
          </p>
          <p
            class="card__Text-sc-s87lnr-4 fABboK"
          >
            An exploration of big cat breeding and its bizarre underworld, populated by eccentric characters.
          </p>
        </div>
      </div>
    </div>
  </div>
  <div
    class="card__Container-sc-s87lnr-1 cVRMWf"
  >
    <p
      class="card__Title-sc-s87lnr-0 eHOFuf"
    >
      Comedies
    </p>
    <div
      class="card__Entities-sc-s87lnr-12 kdrVnm"
    >
      <div
        class="card__Item-sc-s87lnr-13 cmuEeG"
      >
        <img
          class="card__Image-sc-s87lnr-14 hgMODG"
          src="/images/series/comedies/the-office/small.jpg"
        />
        <div
          class="card__Meta-sc-s87lnr-11 beFYBe"
        >
          <p
            class="card__SubTitle-sc-s87lnr-3 zPluP"
          >
            The Office
          </p>
          <p
            class="card__Text-sc-s87lnr-4 fABboK"
          >
            A motley group of office workers go through hilarious misadventures at the Scranton, Pennsylvania, branch of the Dunder Mifflin Paper Company.
          </p>
        </div>
      </div>
    </div>
  </div>
</div>
`;
